app/Http/Controllers/UserController.php

```diff
@@ -0,0 +1,213 @@
+<?php
+
+namespace Suyabay\Http\Controllers;
+
+use Suyabay\User;
+use Suyabay\Role;
+use Suyabay\Invite;
+use Suyabay\Http\Requests;
+use Illuminate\Http\Request;
+use Illuminate\Mail\Mailer as Mail;
+use Suyabay\Http\Controllers\Controller;
+
+class UserController extends Controller
+{
+    protected $mail;
+    protected $response;
+
+    public function __construct(Mail $mail)
+    {
+        $this->mail = $mail;
+    }
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $users = User::orderBy('id', 'asc')->paginate(10);
+
+        return view('dashboard.pages.user', compact('users'));
+    }
+
+    /**
+     * Insert invitation details to invites tablee
+     *
+     * @param  Request $request
+     */
+    public function createInvite(Request $request)
+    {
+        $createInvite = Invite::firstOrCreate([
+            'username'  => $request->username,
+            'role_id'  => $request->user_role,
+            'token'   => $request->_token
+        ]);
+        if ($createInvite)
+        {
+            return $this->getInviteeEmail($request);
+        }
+        else
+        {
+            $this->response =
+            [
+                'message' => 'Error sending Invites',
+                'status_code' => 502
+            ];
+        }
+        return $this->response;
+    }
+
+    /**
+     * Get Invitee email and process mail
+     *
+     * @param  Request $request
+     */
+    public function getInviteeEmail(Request $request)
+    {
+        $email = User::where('username', $request->username)->first();
+        if ($email)
+        {
+            $this->response = $this->sendMail($request, $email);
+        }
+        return $this->response;
+    }
+
+    /**
+     * Send Invitation mail to selected user
+     *
+     * @param  $username
+     * @param  $token
+     * @param  $email
+     *
+     * @return integer
+     */
+    public function sendMail(Request $request, $email)
+    {
+        $mailSent = $this->mail->send('emails.adminInvite', ['username' => $request->username, 'token' => $request->_token], function ($message) use ($email) {
+                $message->from(getenv('SENDER_ADDRESS'), getenv('SENDER_NAME'));
+                $message->to($email->email)->subject('Suyabay Invitation');
+            });
+
+        if ($mailSent)
+        {
+            $this->response =
+            [
+                'message' => 'Invitation was sent successfully',
+                'status_code' => 500
```

Comment:
Okay, should the `status code` be an abstract of `200` i.e `201`, `202`,  `203` for all the successful response?